packages/core/src/tracing/cls.js

```diff
@@ -282,9 +283,15 @@ function startSpan(spanName, kind, traceId, parentSpanId, w3cTraceContext) {
 
   // If the client code has specified a trace ID/parent ID, use the provided IDs.
   if (traceId) {
-    span.t = traceId;
+    // The incoming trace ID/span ID from an upstream tracer could be shorter than the standard length. Some of our code
+    // (in particular, the binary Kafka trace correlation header X_INSTANA_C) assumes the standard length. We normalize
+    // both IDs here by left-padding with 0 characters.
+
+    // Maintenance note (128-bit-trace-ids): When we switch to 128 bit trace IDs, we need to left-pad the trace ID to 32
+    // characters instead of 16.
+    span.t = leftPad(traceId, 16);
```

Comment:
The left pad here (and the one two lines below) are the actual fix.

Reply:
Question: Was wondering if there is a case that the incoming traceID is longer than 16.

Reply:
Good question.

Yes, there are cases, in particular 128 bit trace IDs, which are 32 characters long. Being able to correctly handle 128 bit trace IDs correctly is an important part of our strategy for migrating to that length. See https://www.notion.so/instana/Migration-To-128-Bit-Trace-IDs-e5a6615416f645f792f5dec96d3e2896 for our internal docs on that. At the moment, no Instana tracer will send a 128 bit trace ID via HTTP etc., but at some point that will change, and of course we need to make sure that our tracers of today are already capable of handling that correctly.

Most of that is handled in https://github.com/instana/nodejs/blob/main/packages/core/src/tracing/tracingHeaders.js#L364-L370.

However, now that you have brought it up, I think there are two cases we might want to think about:
* The `tracingHeaders.limitTraceId` function checks for length >= 32. Trace IDs with a length > 16 and < 32 will not be limited. (I don't think these occur out in the wild but well, you never know.)
* Not every incoming trace ID goes through `tracingHeaders.limitTraceId`. For example, for grpcJs.

These are definitely things we should check on. I lean towards doing that in a different PR, and focussing on trace IDs shorter than 16 characters in this PR.